packages/firestore/src/remote/datastore.ts

```diff
@@ -254,12 +258,18 @@ export async function invokeRunAggregationQueryRpc(
     ProtoRunAggregationQueryRequest,
     ProtoRunAggregationQueryResponse
   >('RunAggregationQuery', parent!, request, /*expectedResponseCount=*/ 1);
-  return (
-    response
-      // Omit RunAggregationQueryResponse that only contain readTimes.
-      .filter(proto => !!proto.result)
-      .map(proto => proto.result!.aggregateFields!)
+
+  // Omit RunAggregationQueryResponse that only contain readTimes.
+  const filteredResult = response.filter(proto => !!proto.result);
+
+  hardAssert(
```

Comment:
should we just assert length equals 1 here, and move the other assertions (`.result` and `.result.aggregateFields` not undefined) into the `fromAggregationResult` function?

Reply:
I now see that these assertions exist in `fromAggregationResult`

Reply:
Thank you. I cleaned this up.